src/lib.rs

```diff
@@ -0,0 +1,1041 @@
+			})
+		});
+
+		if !con_success.load(Ordering::Acquire) {
+			return Err(Error::ConnectionFailed);
+		}
+
+		let user_config = UserConfig {
+			channel_handshake_limits: ChannelHandshakeLimits {
+				// lnd's max to_self_delay is 2016, so we want to be compatible.
+				their_to_self_delay: 2016,
+				..Default::default()
+			},
+			channel_handshake_config: ChannelHandshakeConfig {
+				announced_channel: announce_channel,
+				..Default::default()
+			},
+			..Default::default()
+		};
+
+		let user_channel_id: u128 = rand::thread_rng().gen::<u128>();
+
+		match self.channel_manager.create_channel(
+			peer_info.pubkey,
+			channel_amount_sats,
+			0,
+			user_channel_id,
+			Some(user_config),
+		) {
+			Ok(_) => {
+				self.peer_store.add_peer(peer_info.clone())?;
+				log_info!(
+					self.logger,
+					"Initiated channel creation with peer {}. ",
+					peer_info.pubkey
+				);
+				Ok(())
+			}
+			Err(e) => {
+				log_error!(self.logger, "Failed to initiate channel creation: {:?}", e);
+				Err(Error::ChannelCreationFailed)
+			}
+		}
+	}
+
+	#[cfg(test)]
+	pub fn sync_wallets(&self) -> Result<(), Error> {
+		let runtime_lock = self.running.read().unwrap();
+		if runtime_lock.is_none() {
+			return Err(Error::NotRunning);
+		}
+		let wallet = Arc::clone(&self.wallet);
+		let tx_sync = Arc::clone(&self.tx_sync);
+		let sync_cman = Arc::clone(&self.channel_manager);
+		let sync_cmon = Arc::clone(&self.chain_monitor);
+		let confirmables = vec![
+			&*sync_cman as &(dyn Confirm + Sync + Send),
+			&*sync_cmon as &(dyn Confirm + Sync + Send),
+		];
+
+		let runtime = runtime_lock.as_ref().unwrap();
+		tokio::task::block_in_place(move || {
+			runtime.tokio_runtime.block_on(async move { wallet.sync().await })
+		})?;
+
+		tokio::task::block_in_place(move || {
+			runtime.tokio_runtime.block_on(async move { tx_sync.sync(confirmables).await })
+		})?;
+
+		Ok(())
+	}
+
+	/// Close a previously opened channel.
+	pub fn close_channel(
+		&self, channel_id: &[u8; 32], counterparty_node_id: &PublicKey,
+	) -> Result<(), Error> {
+		self.peer_store.remove_peer(counterparty_node_id)?;
+		match self.channel_manager.close_channel(channel_id, counterparty_node_id) {
+			Ok(_) => Ok(()),
+			Err(_) => Err(Error::ChannelClosingFailed),
+		}
+	}
+
+	/// Send a payement given an invoice.
+	pub fn send_payment(&self, invoice: Invoice) -> Result<PaymentHash, Error> {
+		let runtime_lock = self.running.read().unwrap();
+		if runtime_lock.is_none() {
+			return Err(Error::NotRunning);
+		}
+
+		let runtime = runtime_lock.as_ref().unwrap();
+
+		// TODO: ensure we never tried paying the given payment hash before
+		let status = match runtime.invoice_payer.pay_invoice(&invoice) {
+			Ok(_payment_id) => {
+				let payee_pubkey = invoice.recover_payee_pub_key();
+				// TODO: is this unwrap safe? Would a payment to an invoice with None amount ever
+				// succeed? Should we allow to set the amount in the interface or via a dedicated
+				// method?
```

Comment:
You would have gotten an `payment::PaymentError::Invoice` error. So you'd call `pay_zero_value_invoice` instead.

Reply:
I don't have a strong opinion in this case, especially if we want an option to pay more than the invoice amount. That said, more generally, I typically would prefer to add a method rather than see a method with a growing list of `Option` parameters where users need to pass `None` for all of them to get the default behavior.

Reply:
What's the plan for zero-amount invoices?